src/client/datascience/kernel-launcher/localKernelFinder.ts

```diff
@@ -130,7 +130,7 @@ export class LocalKernelFinder implements ILocalKernelFinder {
                 ) || 'root';
 
             // If we have not already searched for this resource, then generate the search
-            if (workspaceFolderId && !this.workspaceToMetadata.has(workspaceFolderId)) {
+            if (!this.workspaceToMetadata.has(workspaceFolderId)) {
```

Comment:
Why did this removal change anything? From the assignment above getWorkspaceFolderIdentifier should return a string. And if the string was '' it should have used 'root', right?

Reply:
Nope, if `worksapceFolderId` is empty, and if user doesn't have a workspace folder open, then it will result in an empty string '', and under that condition we don't have anything, and later in the code we fall over.

This is where things break (line 153)
```return await this.workspaceToMetadata.get(workspaceFolderId)!;```

Reply:
The use of `!` is the root cause.

Reply:
Doesn't the || 'root' in line 130 assure that we'd have 'root' instead of ''? '' should be falsy. Or does the value of 'root' break in line 153?

Reply:
Oh yes, let me double check that

Reply:
Yup, i think with the changes Rich has made the original issue should be resolved.